### DISCUSSION TITLE
[Enhancement] Decouple windowing and basic eventing?

### DISCUSSION THREAD

**Discussion by BreyerW** [dotnet/maui#78] [General]
Did you consider decoupling windowing and basic events from maui in a way that would allow us to get windowing and mouse/keyboard/windows/pads etc without luggage thats is whole maui/xaml?

This would benefit net ecosytem at large not just those that gonna use maui eg avalonia could hypothetically consider moving to it if it was found better than what they have now (eg easier porting to new platforms thanks to microsoft and whole net community involvement which is certainly bigger than just avalonia community)

same apply to any other open source uis out there

Another long term benefit would be easier experimenting with maui or windowing itself (because of shim created in maui itself that could hide changes to separate package) or easier investing on new, better xplat ui than maui in future (lets be real microsft loves to dump or abandon projects after 5-20 years, this effort could at least save windowing for future further than that since to me this part is basically boilerplate code rather than something you can innovate on maybe with exception of inputs)

Basically i wish c# world had something akin to [glfw](https://github.com/glfw/glfw) it has very little outside of creating and managing windows+"basic" events (unlike sdl2 which tries to support creating games mainly) making it perfect for mixing and matching with various libraries but i would like to have it written in c# as much as possible with as little code written in c/c++ as possible and wider platform selection in future, mobile and/or console in particular
